psst-gui/src/data/mod.rs

```diff
@@ -180,11 +182,19 @@ impl AppState {
 
 impl AppState {
     pub fn queued_entry(&self, item_id: ItemId) -> Option<QueueEntry> {
-        self.playback
-            .queue
-            .iter()
-            .find(|queued| queued.item.id() == item_id)
-            .cloned()
+        if let Some(queued) = self.playback.queue.iter().find(|queued| queued.item.id() == item_id).cloned() {
+            return Some(queued);
+        } else if let Some(queued) = self.added_queue.iter().find(|queued| queued.item.id() == item_id).cloned() {
+            return Some(queued);
+        } else {
+            None
+        }
+    }
```

Comment:
Hmm, can you justify why we keep a separate queue from the contextual playback queue? I just maybe want to think about the justification.

Reply:
I believe that having a separate queue allows the program to avoid removing the user queued songs. When the user clicks on a different playlist, a new queue is created entirely meaning the main song queue gets rewritten, in my opinion showing the need for a separate queue.

This approach also ensures that the song contexts are passed correctly. Otherwise, an error occurs where only the audio loads, but none of the data is found in the main fetched queue.

From a UI perspective, having a separate user-added queue allows for a more minimal display of just the user-added songs. Alternatively, it provides the option to have two different scrollable contexts where the user can view both queues separately.

Please let me know if this made sense and thank you so much for your feedback :)

Reply:
Thanks for the explanation, those were my thoughts as well just wanted some clarity on the arch decision for this PR.